src/org/exist/xquery/functions/transform/ExistUriResolver.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *  eXist Open Source Native XML Database
+ *  Copyright (C) 2001-2016 The eXist Project
+ *  http://exist-db.org
+ *
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2
+ *  of the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery.functions.transform;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.exist.dom.persistent.BinaryDocument;
+import org.exist.dom.persistent.DocumentImpl;
+import org.exist.security.Permission;
+import org.exist.security.PermissionDeniedException;
+import org.exist.storage.DBBroker;
+import org.exist.xmldb.XmldbURI;
+
+import javax.xml.transform.Source;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.URIResolver;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+
+/**
+ * Implementation of URIResolver which
+ * will resolve paths from the eXist database
+ */
+public class EXistURIResolver implements URIResolver {
+    private static final Logger LOG = LogManager.getLogger(EXistURIResolver.class);
+
+    final DBBroker broker;
+    final String basePath;
+
+    public EXistURIResolver(final DBBroker broker, final String docPath) {
+        this.broker = broker;
+        this.basePath = docPath;
+        LOG.debug("EXistURIResolver base path set to " + basePath);
+    }
+
+    /**
+     * Simplify a path removing any "." and ".." path elements.
+     * Assumes an absolute path is given.
+     */
+    private String normalizePath(final String path) {
```

Comment:
@dizzzz Okay, if it is unused should we remove it? If not perhaps switch to using the one I extended?